src/test/java/com/aws/greengrass/device/iot/CertificateRegistryTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.device.iot;
+
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+class CertificateRegistryTest {
+    @Mock
+    private IotAuthClient mockIotAuthClient;
+
+    @Captor
+    private ArgumentCaptor<String> certPemCaptor;
+
+    private CertificateRegistry registry;
+
+    @BeforeEach
+    void beforeEach() {
+        registry = new CertificateRegistry(mockIotAuthClient);
+    }
+
+    @AfterEach
+    void afterEach() {
+        registry.clear();
+    }
+
+    @Test
+    void GIVEN_certificatePem_and_cloudProperResponse_WHEN_getIotCertificateIdForPem_THEN_certificateIdReturned() {
+        String mockCertPem = "certificatePem";
```

Comment:
Nit: can pull these out as constants since they're used in all of these tests

Reply:
updated